From 46c1b13fea155032e16c6763deee69cf4c63864a Mon Sep 17 00:00:00 2001
From: Chen Wandun <chenwandun@huawei.com>
Date: Mon, 18 Apr 2022 14:34:22 +0800
Subject: [PATCH 3594/4585] [Huawei] Revert "mm/page_cache_limit: do
 shrink_page_cache when adding page to page cache"

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187317

--------------------------------

This reverts commit 9fea105d74885fef299b43e6734d19ba8921242e.
This feature will be reimplement.

Signed-off-by: Chen Wandun <chenwandun@huawei.com>
---
 include/linux/page_cache_limit.h | 2 --
 include/linux/pagemap.h          | 2 --
 mm/filemap.c                     | 2 --
 3 files changed, 6 deletions(-)

diff --git a/include/linux/page_cache_limit.h b/include/linux/page_cache_limit.h
index 2df08a0604d8..7906b12af947 100644
--- a/include/linux/page_cache_limit.h
+++ b/include/linux/page_cache_limit.h
@@ -19,11 +19,9 @@ unsigned long __shrink_node_page_cache(int nid, gfp_t mask,
 		unsigned long nr_to_reclaim, enum page_cache_reclaim_flag flag);
 void kpagecache_limitd_stop(int nid);
 int kpagecache_limitd_run(int nid);
-void wakeup_all_kpagecache_limitd(void);
 #else
 static inline void kpagecache_limitd_stop(int nid) {}
 static inline int kpagecache_limitd_run(int nid) { return 0; }
-static inline void wakeup_all_kpagecache_limitd(void) {}
 #endif
 
 #endif
diff --git a/include/linux/pagemap.h b/include/linux/pagemap.h
index dbb25f1dc2e9..0bfa9cce6589 100644
--- a/include/linux/pagemap.h
+++ b/include/linux/pagemap.h
@@ -15,7 +15,6 @@
 #include <linux/bitops.h>
 #include <linux/hardirq.h> /* for in_interrupt() */
 #include <linux/hugetlb_inline.h>
-#include <linux/page_cache_limit.h>
 
 struct pagevec;
 
@@ -778,7 +777,6 @@ static inline int add_to_page_cache(struct page *page,
 {
 	int error;
 
-	wakeup_all_kpagecache_limitd();
 	__SetPageLocked(page);
 	error = add_to_page_cache_locked(page, mapping, offset, gfp_mask);
 	if (unlikely(error))
diff --git a/mm/filemap.c b/mm/filemap.c
index c0bcbd278d3c..05f102130fb0 100644
--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -42,7 +42,6 @@
 #include <linux/psi.h>
 #include <linux/ramfs.h>
 #include <linux/page_idle.h>
-#include <linux/page_cache_limit.h>
 #include "internal.h"
 
 #define CREATE_TRACE_POINTS
@@ -929,7 +928,6 @@ int add_to_page_cache_lru(struct page *page, struct address_space *mapping,
 	void *shadow = NULL;
 	int ret;
 
-	wakeup_all_kpagecache_limitd();
 	__SetPageLocked(page);
 	ret = __add_to_page_cache_locked(page, mapping, offset,
 					 gfp_mask, &shadow);
-- 
2.17.1

